sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -378,25 +378,20 @@ dmlStatementNoWith
     ;
 
 queryNoWith
-    : queryTerm queryOrganization                                               #noWithQuery
-    | fromClause selectStatement+                                               #queryWithFrom
+    : queryTerm queryOrganization
     ;
 
 queryOrganization
     : (ORDER BY order+=sortItem (',' order+=sortItem)*)?
       (CLUSTER BY clusterBy+=expression (',' clusterBy+=expression)*)?
       (DISTRIBUTE BY distributeBy+=expression (',' distributeBy+=expression)*)?
       (SORT BY sort+=sortItem (',' sort+=sortItem)*)?
-      windows?
+      windowClause?
       (LIMIT (ALL | limit=expression))?
     ;
 
 multiInsertQueryBody
-    : insertInto selectStatement
-    ;
-
-selectStatement
-    : querySpecification queryOrganization
+    : insertInto fromStatementBody
```

Comment:
The clause can be either select/transform/map/reduce etc. Thus rename this to `fromStatementBody`.